signature/internal/rekor_set.go

```diff
@@ -40,29 +41,30 @@ type UntrustedRekorPayload struct {
 // A compile-time check that UntrustedRekorSET implements json.Unmarshaler
 var _ json.Unmarshaler = (*UntrustedRekorSET)(nil)
 
-// UnmarshalJSON implements the json.Unmarshaler interface
-func (s *UntrustedRekorSET) UnmarshalJSON(data []byte) error {
-	err := s.strictUnmarshalJSON(data)
-	if err != nil {
-		if formatErr, ok := err.(JSONFormatError); ok {
-			err = NewInvalidSignatureError(formatErr.Error())
-		}
+// ConvertFormatError converts JSONFormatError to NewInvalidSignatureError.
+// All other errors are returned as is.
+func ConvertFormatError(err error) error {
```

Comment:
“convert” does not really say what this does. `JSONFormatToInvalidSignatureError`, or something?

Reply:
I partially agree, but the function itself is quite small and easy to understand, plus it is documented, so I did not feel a need to try to place the entirely of what it does into its name. To me, oftentimes smaller names are way more readable (i.e. when a human being reads it from the screen).

Reply:
In particular, I find `JSONFormatToInvalidSignatureError` way too long to be comprehensible.

Reply:
The `signature` area is critical and culturally very paranoid, and clarity is important. Short names work in small contexts, but this is used across subpackages.

Alternatively, I’m inclined to say that if we can’t think of a good name for a function, maybe it shouldn’t be a separate function.

Reply:
Opened #2518.